localstack-core/localstack/utils/analytics/usage.py

```diff
@@ -49,6 +49,59 @@ def aggregate(self) -> dict:
         return self.state
 
 
+class UsageMultiSetCounter:
+    """
+    Use this counter to count occurrences of unique values for multiple dimensions.
+    This dynamically creates UsageSetCounters and should be used with care (i.e., with limited keys).
+
+    Example:
+
+    my_feature_counter = UsageMultiSetCounter("pipes:invocation")
+    my_feature_counter.record("aws:sqs", "aws:lambda")
+    my_feature_counter.record("aws:sqs", "aws:lambda")
+    my_feature_counter.record("aws:sqs", "aws:stepfunctions")
+    my_feature_counter.record("aws:kinesis", "aws:lambda")
+    aggregate is implemented for each counter individually
+
+    my_feature_counter.aggregate() is available for testing purposes:
+    {
+       "aws:sqs": {
+         "aws:lambda": 2,
+         "aws:stepfunctions": 1,
+       },
+       "aws:kinesis": {
+         "aws:lambda": 1
+       }
+    }
+    """
+
+    namespace: str
+    _counters: dict[str, UsageSetCounter]
+
+    def __init__(self, namespace: str):
+        self._counters = {}
+        self.namespace = namespace
+
+    def record(self, key: str, value: str):
+        namespace = f"{self.namespace}:{key}"
+        # We cannot use setdefault here because Python always instantiates a new UsageSetCounter,
+        # which overwrites the collector_registry
+        if namespace in self._counters:
+            set_counter = self._counters[namespace]
+        else:
+            set_counter = UsageSetCounter(namespace)
+            self._counters[namespace] = set_counter
```

Comment:
This is not thread safe, we might unnecessarily instantiate some usagesetcounters with the same namespace here, which might override each other in an unprecedented way. This could for example happen in multiple event source mappings in multiple thread.

Since we cannot use setdefault, I'd rather just add a lock here.